BLUESPAWN-client/src/util/filesystem/FileSystem.cpp

```diff
@@ -660,4 +761,66 @@ namespace FileSystem{
 		} while(MoveToNextFile());
 		return toRet;
 	}
+
+	std::optional<Permissions::Owner> Folder::GetFolderOwner() const {
+		if (!bFolderExists) {
+			LOG_ERROR("Can't get owner of nonexistent folder " << FolderPath);
+			return std::nullopt;
+		}
+		PSID psOwnerSID = NULL;
+		PISECURITY_DESCRIPTOR pDesc = NULL;
+		if (GetNamedSecurityInfoW((LPWSTR) FolderPath.c_str(), SE_FILE_OBJECT, OWNER_SECURITY_INFORMATION, &psOwnerSID, nullptr, nullptr, nullptr, reinterpret_cast<PSECURITY_DESCRIPTOR*>(&pDesc)) != ERROR_SUCCESS) {

```

Comment:
Not gonna reject the commit over this (and I'm guilty of this too sometimes), but C style casts in c++ are bad practice. Use reinterpret_cast, const_cast, dynamic_cast or static_cast if possible